docs/core/compatibility/wcf-client/6.0/duplex-synchronization-context.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "Breaking change: Duplex contract captures synchronization context"
+description: Learn about the breaking change in WCF Client 6.0 where the duplex contract now captures synchronization context when it's opened.
+ms.date: 04/06/2023
+---
+# Duplex contract captures synchronization context
+
+The duplex contract now captures the synchronization context when it's opened and dispatches callbacks to the captured context.
+
+## Previous behavior
+
+In previous versions, the duplex contract failed to capture synchronization context when the channel factory was opened.
+
+## New behavior
+
+Starting in WCF Client 6.0 RC, the duplex channel captures synchronization context and dispatches callbacks to the captured context. This behavior matches that of .NEt Framework.
+
+## Version introduced
+
+WCF Client 6.0 RC
+
+## Type of breaking change
+
+This change is a [behavioral change](../../categories.md#behavioral-change).
+
+## Reason for change
+
+This change was made to [fix a bug](https://github.com/dotnet/wcf/issues/4946) where the client hung when making an asynchronous call followed by a synchronous call. The new behavior matches .NET Framework functionality.
```

Comment:
That wasn't the bug for this change. It was an internal find which was reported directly to me. An issue was never opened for it. The internal problem was a callback was instantiating a STA COM object and as the initial callback and subsequent callbacks weren't being dispatched to a custom SynchronizationContext, the callbacks were happening on the wrong thread.